File: C/pr_5.c
#include <stdio.h>

void insert_element(int arr[], int *size, int element, int k) {
    int i;

    // Shift elements to the right to make space for the new element
    for (i = *size - 1; i >= k; i--) {
        arr[i + 1] = arr[i];
    }

    // Insert the new element at the given location
    arr[k] = element;

    // Increment the size of the array
    (*size)++;
}

int main() {
    int my_array[100] = {1, 2, 3, 4, 5};
    int size = 5;
    int new_element = 6;
    int insert_location = 2;

    printf("Original array: ");
    for (int i = 0; i < size; i++) {
        printf("%d ", my_array[i]);
    }
    printf("\n");

    insert_element(my_array, &size, new_element, insert_location);

    printf("Array after insertion: ");
    for (int i = 0; i < size; i++) {
        printf("%d ", my_array[i]);
    }
    printf("\n");

    return 0;
}

File: Python/pr_5.py
def insert_element(arr, element, k):
    arr.append(None)  # Add a placeholder element at the end

    # Shift elements to the right to make space for the new element
    for i in range(len(arr) - 1, k, -1):
        arr[i] = arr[i - 1]

    # Insert the new element at the given location
    arr[k] = element

    return arr

my_array = [1, 2, 3, 4, 5]
new_element = 6
insert_location = 2

print("Original array:", my_array)

updated_array = insert_element(my_array, new_element, insert_location)

print("Array after insertion:", updated_array)
